shared/validate/validate.go

```diff
@@ -787,3 +788,31 @@ func IsRequestURL(value string) error {
 
 	return nil
 }
+
+// IsCloudConfig checks value is a valid cloud-config.
+func IsCloudConfig(value string) error {
+	lines := strings.SplitN(value, "\n", 2)
+
+	if lines[0] != "#cloud-config" {
+		return fmt.Errorf("Missing cloud-config header")
+	}
```

Comment:
We shouldn't do that. There are plenty of other valid formats for cloud-init other than `#cloud-config`.

Anything other than `#cloud-config` should be considered valid and be left alone.

Reply:
I've updated the PR to only validate YAML if the `#cloud-config` header is present.